From c737a3cd87a122fc7f85307e9b6ea17c5bf09f57 Mon Sep 17 00:00:00 2001
From: Markus Trippelsdorf <markus@trippelsdorf.de>
Date: Tue, 19 Sep 2017 12:37:24 +0200
Subject: [PATCH] VFS: Handle lazytime in do_mount()
Patch-mainline: Submitted, https://patchwork.kernel.org/patch/9958653/
References: boo#1068256

Since commit e462ec50cb5fa ("VFS: Differentiate mount flags (MS_*) from
internal superblock flags") the lazytime mount option didn't get passed
on anymore.

Fix the issue by handling the option in do_mount().

Signed-off-by: Markus Trippelsdorf <markus@trippelsdorf.de>
Reviewed-by: Lukas Czerner <lczerner@redhat.com>
Acked-by: Goldwyn Rodrigues <rgoldwyn@suse.com>
---
 fs/namespace.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/namespace.c b/fs/namespace.c
index 23cdf6c62895..9c254f515fb6 100644
--- a/fs/namespace.c
+++ b/fs/namespace.c
@@ -2826,7 +2826,8 @@ long do_mount(const char *dev_name, const char __user *dir_name,
 			    SB_DIRSYNC |
 			    SB_SILENT |
 			    SB_POSIXACL |
-			    SB_I_VERSION);
+			    SB_I_VERSION |
+			    SB_LAZYTIME);
 
 	if (flags & MS_REMOUNT)
 		retval = do_remount(&path, flags, sb_flags, mnt_flags,
-- 
2.13.6

